### ISSUE TITLE
Possible Problem with Time

### ISSUE THREAD

**Issue by wcolling** [obones/openhab-binding-openmeteo#1]
I have added an item named localTodayMaxTemperature and I populate it in a rule with:

`localTodayMaxTemperature.postUpdate(OpenMeteo_local_forecast_Maximum_outdoor_temperature_Daily.historicState(now, "influxdb").state)`

I am on the west coast of the US and at 8PM OpenMeteo_local_forecast_Maximum_outdoor_temperature_Daily.historicState(now, "influxdb").state returns tomorrow's maximum, not today's.

I see that the binding's call to OpenMeteo includes timezone=UTC. Shouldn't this be timezone=auto or, at least, an option to allow me to set the timezone?

**Comment by obones** [comment]
Hello,

The binding is using `Instant.ofEpochSeconds` as can be seen [here](https://github.com/obones/openhab-binding-openmeteo/blob/main/src/main/java/com/obones/binding/openmeteo/internal/handler/OpenMeteoForecastThingHandler.java#L936)

The [documentation](https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html#ofEpochSecond-long-) for that method says it's working from the `1970-01-01T00:00:00Z` which to me is consistent with requiring times in the UTC timezone, also known as the Zulu time zone.

Could it be that the `historicState` expects another type than just what `now` is returning? I can't find a reference but I have a memory about using `ZonedDateTime` in the JS scripts specifically to avoid such issues.

**Comment by wcolling** [comment]
Here is the problem. The daily values are persisted to influx. The timestamps are in GMT. Here are my current values:

```
> select * from OpenMeteo_local_forecast_Maximum_outdoor_temperature_Daily
name: OpenMeteo_local_forecast_Maximum_outdoor_temperature_Daily
time                item                                                       value
----                ----                                                       -----
1714953600000000000 OpenMeteo_local_forecast_Maximum_outdoor_temperature_Daily 75.4754
1715040000000000000 OpenMeteo_local_forecast_Maximum_outdoor_temperature_Daily 74.2154018
1715126400000000000 OpenMeteo_local_forecast_Maximum_outdoor_temperature_Daily 76.9154018
1715212800000000000 OpenMeteo_local_forecast_Maximum_outdoor_temperature_Daily 81.2354
1715299200000000000 OpenMeteo_local_forecast_Maximum_outdoor_temperature_Daily 83.5817
1715385600000000000 OpenMeteo_local_forecast_Maximum_outdoor_temperature_Daily 83.9417
1715472000000000000 OpenMeteo_local_forecast_Maximum_outdoor_temperature_Daily 85.7417
1715558400000000000 OpenMeteo_local_forecast_Maximum_outdoor_temperature_Daily 84.9317
1715644800000000000 OpenMeteo_local_forecast_Maximum_outdoor_temperature_Daily 83.2217018
1715731200000000000 OpenMeteo_local_forecast_Maximum_outdoor_temperature_Daily 77.4617

The data is for May 6, 2024 12:00 AM GMT through May 15, 2024 12:00 AM GMT.

When I request the daily value for my today:
Now =2024-05-08T17:12:00.864360068-07:00[America/Los_Angeles]

the date is converted to GMT and the associated state is returned. This means when I request for the 8th after 5PM, I get the state for the 9th. Obviously, not what I want.

When I first encountered this using the OpenMeteo API, I found that switching to timestamp=auto fixed my problem with the returned data.

Currently, everything seems to be working correctly (openhab, the binding, and persistence), but I still get the wrong result. I was hoping that storing the data in local time (timestamp=auto) would get around the problem, but now I am not sure.

**Comment by obones** [comment]
Here is how I retrieve the timeseries data from one of my JS rules:

```js
const now = time.toZDT();
const forecastStart = now.withHour(0).withMinute(0).withSecond(0);
const forecastEnd = forecastStart.plusDays(6);
          
const dailyWMOCodeHistoricItems = dailyWMOCodeItem.history.getAllStatesBetween(forecastStart, forecastEnd);
```

Do you define `now` the same way that I do?

**Comment by wcolling** [comment]
I was able to query the daily data by adding and subtracting UTC offset to the local date to get around the skewed daily data in the persisted data, but your suggestion is far easier. I just need to know if the weather location is before or following UTC. For locations ahead of UTC the corresponding code is

`var queryDate = now.withHour(23).withMinute(59)`

I tested retrieving the Hourly data and it works as expected. I can query using the local datetime or with the corresponding UTC time.
